File: WorldFeed/Client/src/app/core/Public/Exceptions/TwitterResponseException.ts
import Exception from "../../../c#-objects/TypeScript.NET-Core/packages/Core/source/Exception";
import {ITwitterResult} from "../../Core/Web/TwitterResult";

export class TwitterResponseException extends Exception {
  constructor(twitterResult: ITwitterResult, message?: string) {
    if (message) {
      super(message);
      this.twitterResult = twitterResult;
    } else {
      this.twitterResult = twitterResult;
    }
  }

  public twitterResult: ITwitterResult;
}

// public TwitterResponseException(ITwitterResult twitterResult)
// {
//     TwitterResult = twitterResult;
// }
//
// public TwitterResponseException(ITwitterResult twitterResult, string message) : base(message)
// {
//     TwitterResult = twitterResult;
// }

File: Sharebook/Server/src/Services/Identity/Sharebook.Identity.API/Application/Exceptions/TwitterResponseException.cs
namespace Sharebook.Identity.API.Application.Exceptions
{
    using System;

    using Sharebook.Identity.API.Application.Web;

    public class TwitterResponseException : Exception
    {
        public ITwitterResult TwitterResult { get; }

        public TwitterResponseException(ITwitterResult twitterResult)
        {
            this.TwitterResult = twitterResult;
        }

        public TwitterResponseException(ITwitterResult twitterResult, string message) : base(message)
        {
            this.TwitterResult = twitterResult;
        }
    }
}
